deployment/terraform/resources/keyvault.tf

```diff
@@ -63,3 +63,15 @@ data "azurerm_key_vault_secret" "backend_app_id" {
   name         = var.backend_api_app_id_secret_name
   key_vault_id = data.azurerm_key_vault.deploy_secrets.id
 }
+
+## Access Policies
+
+# The Key Vault service principal should be able to get secrets.
+resource "azurerm_key_vault_access_policy" "task-sp-get-secrets" {
```

Comment:
Not sure if this was done manually previously, but this should ensure that the `kv` Service Principal can access secrets, so that templating works. Without this, the argo workflow pod failed trying to access those secrets.